GitUI/UserControls/RevisionGrid.cs

```diff
@@ -2761,18 +2762,33 @@ private void CheckUncommitedChanged(string filtredCurrentCheckout)
                 stageCount = "";
             }
 
-            //Add working directory as virtual commit
+            var userName = Module.GetEffectiveSetting(SettingKeyString.UserName);
+            var userEmail = Module.GetEffectiveSetting(SettingKeyString.UserEmail);
+
+            // Add working directory as virtual commit
             var workingDir = new GitRevision(Module, GitRevision.UnstagedGuid)
             {
+                Author = userName,
+                AuthorDate = DateTime.MaxValue,
+                AuthorEmail = userEmail,
+                Committer = userName,
+                CommitDate = DateTime.MaxValue,
+                CommitterEmail = userEmail,
                 SubjectCount = unstageCount,
```

Comment:
SubjectCount is special, separate first or last in the block.

Reply:
I am afraid I don't understand what you mean.

Reply:
Information for a commit is static, except BuildInfo and SubjectCount (that is merged to Subject). Separate SubjectCount from the static fields.

Reply:
Right now `GitRevision` depends on an instance of `GitModule` which isn't static. So the artificial revision instances need to be context aware...
`CheckUncommitedChanged` is called once whenever the graph is refreshed, so performance hit is probably minimal.

Reply:
Yes, I suggest leave it as is. Git revision impact should be low (that the content is not static supports my point, but we already have had this discussion). 

Separate CheckUncommitedChanged is removed (shared with commit button) in #4209